xtuner/v1/train/trainer.py

```diff
@@ -461,6 +464,9 @@ def __init__(
 
         setup_prober_list(self.exp_dir, self._profile_step, self._engine.model, prober_list)
 
+        if self._internal_metrics_interval:
+            self.metrics_recorder = InternalMetricsRecorder(self._engine)
```

Comment:
Suggest making it default to None.  It's not a good idea to define a `not requried` attribute in class. For example:

```python
class Foo:
    def __init__(self, a: int | None = 1):
        if a is not None:
            self.a = a


    def get(self) -> int:
        return self.a
```

In this case, static checker like mypy will not raise any error, although the `get` method will raise an unexpected error when `a` is None, besides, the typehint mismatched.

If you make `a` default to None, mypy could find the error